corehq/ex-submodules/pillowtop/tasks.py

```diff
@@ -0,0 +1,63 @@
+from celery.schedules import crontab
+from celery.task import periodic_task
+from django.conf import settings
+
+from corehq.util.datadog.gauges import datadog_gauge
+from corehq.util.decorators import serial_task
+from pillowtop.utils import get_all_pillows_json
+
+
+@periodic_task(run_every=crontab(minute="*/5"), queue=settings.CELERY_PERIODIC_QUEUE)
+def run_pillow_datadog_metrics():
+    pillow_datadog_metrics.delay()
+
+
+@serial_task('pillow-datadog-metrics', timeout=30 * 60, queue=settings.CELERY_PERIODIC_QUEUE, max_retries=0)
+def pillow_datadog_metrics():
+    def _is_couch(pillow):
+        # text is couch, json is kafka
+        return pillow['seq_format'] == 'text'
+
+    pillow_meta = get_all_pillows_json()
+
+    for pillow in pillow_meta:
+        tags = [
+            'pillow_name:{}'.format(pillow['name']),
+            'feed_type:{}'.format('couch' if _is_couch(pillow) else 'kafka')
+        ]
+
+        for topic_name, offset in pillow['offsets'].items():
+            if _is_couch(pillow):
+                assert isinstance(pillow['seq'], int)
+                assert len(pillow['offsets']) == 1
+                tags_with_topic = tags + ['topic:{}'.format(topic_name)]
+                processed_offset = pillow['seq']
+            else:
+                assert isinstance(pillow['seq'], dict)
+                if not pillow['seq']:
+                    # this pillow has never been initialized.
+                    # (custom pillows on most environments)
+                    continue
+                assert len(pillow['offsets']) == len(pillow['seq'])
+                topic, partition = topic_name.split(',')
+                tags_with_topic = tags + ['topic:{}-{}'.format(topic, partition)]
+                processed_offset = pillow['seq'][topic_name]
```

Comment:
@emord is it useful to have these broken down by partition? I think we might get more use out of it if we combined all the partitions together. What do you think?

Reply:
I think it would be good to these to two separate tags for topic and partition. We get use use out of the partition because it would allow seeing any one process that is behind. (This would be less of an issue if we upgraded kafka so that we could use it to rebalance partitions to other processes)